packages/playwright-core/src/utils/isomorphic/locatorGenerators.ts

```diff
@@ -194,7 +187,33 @@ function innerAsLocators(factory: LocatorFactory, parsed: ParsedSelector, isFram
     let locatorPartWithEngine: string | undefined;
     if (['xpath', 'css'].includes(part.name)) {
       const selectorPart = stringifySelector({ parts: [part] }, /* forceEngineName */ true);
-      locatorPartWithEngine = factory.generateLocator(base, locatorType, selectorPart);
+      locatorPartWithEngine = factory.generateLocator(base, 'default', selectorPart);
+    }
+
+    if (nextPart && nextPart.name === 'internal:control' && (nextPart.body as string) === 'enter-frame') {
```

Comment:
After looking again, I see that we might be missing some cases. For example, `page.getByTitle('iframe title').contentFrame()` - does it correctly handle this case?

Reply:
Great catch, it does not. I've revised the approach in https://github.com/microsoft/playwright/pull/33208/commits/e1e9b665517a526087fffb9f8edae02bdb9c0912 to go back one produced token and update that in retrospect. This makes the change much smaller as well, really liking it. Makes the code a little more hairy though, because it can now change already-produced tokens.